### ISSUE TITLE
Cannot assign to read only property 'cache' of true

### ISSUE THREAD

**Issue by bricss** [gruntjs/grunt-contrib-uglify#318]
Getting this error after update to v0.9.0.

```
Running "uglify:main" (uglify) task
[TypeError: Cannot assign to read only property 'cache' of true]
```

Config:

```js
uglify : {
  main : {
    files : [{
      cwd : '<%= meta.rootPath %><%= meta.srcPath %>/js/',
      dest : '<%= meta.rootPath %><%= meta.srcPath %>/js/',
      expand : true,
      ext : '.min.js',
      src : ['*.js', '!*.min.js']
    }]
  },
  options : {
    banner : '<%= banner.join("\\n") %>',
    beautify : false,
    compress : true,
    mangle : true,
    report : 'gzip'
  }
}
```

**Comment by XhmikosR** [comment]
Yeah, confirmed :/

**Comment by jrhite** [comment]
Ugh...so trivial I missed it...sorry.

The changes I added assume/require options.mangle is an object, not just a simple boolean value. If options.mangle === false there is no problem, but if options.mangle === true the bug is triggered.

I created [PR 319](https://github.com/gruntjs/grunt-contrib-uglify/pull/319) to fix the problem and added a test too. It checks if options.mangle is an object or not, and if not converts it to an object.
